Changes to allow models with non-singular variable names (fixes #1710)

gweesip opened the pull request:
get_minimal_adjustment_set now wraps variables in a list in call to get_proper_backdoor_graph, which expects array-like arguments. This prevents models with variable names that aren't exclusively single letters from raising errors.

Also added example to docstring to demonstrate usage.

### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1710

### List of changes to the codebase in this pull request
- arguments passed to get_proper_backdoor_graph are now wrapped in lists
- added example to docstring
-

gweesip committed:
Changes to allow models with non-singular variable names (fixes #1710)

get_minimal_adjustment_set now wraps variables in a list in call to
get_proper_backdoor_graph, which expects array-like arguments. This
prevents models with variable names that aren't exclusively single
letters from raising errors.

Also added example to docstring to demonstrate usage.

ankurankan committed:
Merge remote-tracking branch 'origin/dev' into bugfix-1710-get-minimal-adjustment-set

ankurankan commented:
@gweesip Thanks a lot for the PR. There were some merge conflicts (I am guessing you branched off an older pgmpy dev branch), but I have fixed that now (please remember to do a git pull on your local branch before making any changes). The changes are good, could you please also add a test for the changes somewhere in this file: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/tests/test_inference/test_CausalInference.py ? Ideally, the test should be failing with the current code, and should get fixed when the changes in this PR gets merged.

gweesip commented:
@ankurankan Thanks for the feedback. I'll add the test as requested.

gweesip committed:
Merge branch 'dev' of https://github.com/gweesip/pgmpy into bugfix-1710-get-minimal-adjustment-set

Updated example in docstring of get_minimal_adjustment_set.

gweesip committed:
Merge branch 'bugfix-1710-get-minimal-adjustment-set' of https://github.com/gweesip/pgmpy into bugfix-1710-get-minimal-adjustment-set

gweesip committed:
Updates to TestAdjustmentSet for issue #1710

test_get_minimal_adjustment_set now includes a test to ensure missing
variables in the model raise ValueError as expected.

New test_issue_1710 tests models with non-singular variables using the
example from the docstring.

ankurankan commented:
@gweesip Looks good. Thanks again for the fix :)
